packages/flutter_tools/lib/src/web/compiler_config.dart

```diff
@@ -0,0 +1,137 @@
+        sourceMaps: defines[kSourceMapsEnabled] == 'true',
+      );
+
+  /// The default optimization level for dart2js.
+  ///
+  /// Maps to [kDart2jsOptimization].
+  static const String kDart2jsDefaultOptimizationLevel = 'O4';
+
+  /// Build environment flag for [optimizationLevel].
+  static const String kDart2jsOptimization = 'Dart2jsOptimization';
+
+  /// Build environment flag for [dumpInfo].
+  static const String kDart2jsDumpInfo = 'Dart2jsDumpInfo';
+
+  /// Build environment flag for [noFrequencyBasedMinification].
+  static const String kDart2jsNoFrequencyBasedMinification =
+      'Dart2jsNoFrequencyBasedMinification';
+
+  /// Build environment flag for [csp].
+  static const String kCspMode = 'cspMode';
+
+  /// Build environment flag for [sourceMaps].
+  static const String kSourceMapsEnabled = 'SourceMaps';
+
+  /// Build environment flag for [nativeNullAssertions].
+  static const String kNativeNullAssertions = 'NativeNullAssertions';
+
+  /// Whether to disable dynamic generation code to satisfy csp policies.
+  final bool csp;
+
+  /// If `--dump-info` should be passed to dart2js.
+  final bool dumpInfo;
+
+  /// Whether the dart2js native null assertions are enabled.
+  final bool nativeNullAssertions;
+
+  // If `--no-frequency-based-minification` should be passed to dart2js
+  // TODO(kevmoo): consider renaming this to be "positive". Double negatives are confusing.
+  final bool noFrequencyBasedMinification;
+
+  /// An override for the dart2js build mode.
+  ///
+  /// Valid values are O1 (lowest, profile default) to O4 (highest, release default).
+  // TODO(kevmoo): consider storing this as an [int] and validating it!
+  final String optimizationLevel;
+
+  /// Whether the dart2js build should output source maps.
+  final bool sourceMaps;
+
+  @override
+  bool get isWasm => false;
+
+  @override
+  Map<String, String> toBuildSystemEnvironment() => <String, String>{
+        kCspMode: csp.toString(),
+        kDart2jsDumpInfo: dumpInfo.toString(),
+        kNativeNullAssertions: nativeNullAssertions.toString(),
+        kDart2jsNoFrequencyBasedMinification: noFrequencyBasedMinification.toString(),
+        kDart2jsOptimization: optimizationLevel,
+        kSourceMapsEnabled: sourceMaps.toString(),
+      };
+
+  /// dart2js arguments to use in both phases: full compile and CFE-only.
+  List<String> toSharedCommandOptions() => <String>[
+        if (nativeNullAssertions) '--native-null-assertions',
+        if (!sourceMaps) '--no-source-maps',
+      ];
+
+  /// dart2js arguments to use in the full compile, but not CFE-only.
+  ///
+  /// Includes the contents of [toSharedCommandOptions].
+  List<String> toCommandOptions() => <String>[
+        ...toSharedCommandOptions(),
+        '-$optimizationLevel',
+        if (dumpInfo) '--dump-info',
+        if (noFrequencyBasedMinification) '--no-frequency-based-minification',
+        if (csp) '--csp',
+      ];
```

Comment:
```suggestion
  List<String> toCommandOptions() => <String>[
    ...toSharedCommandOptions(),
    '-$optimizationLevel',
    if (dumpInfo) '--dump-info',
    if (noFrequencyBasedMinification) '--no-frequency-based-minification',
    if (csp) '--csp',
  ];
```

Reply:
Seems so silly to spend human brain power on whitespace...